docs/spec/interdomain.md

```diff
@@ -0,0 +1,32 @@
+Interdomain NSM
```

Comment:
This is an important feature, please add more explanation here. For example you can answer witha s much details as possible to the following questions:

 - How does one setup the 2 clusters? Any specifics besides using kubeadm to deploy and then helm to install NSM. What else I need to set in order for the interdomain to work?
 - is there a central "registry" where I can see which clusters are connected? Or are we loosely coupled?
 - who sets up the DNS records?
 - how is the client supposed to know in which domain the service lives? Is this a "first step, temporary solution"? Will there be a "domain agnostic" service resolve at some point?
 - When a service is registered, how this registration gets distributed to the other clusters
 - have you tested a service across "many" clusters (i.e. >2)?
 - will we be able to do hybrid cloud solutions with this? Like vagrant deployment talking to GKE hosted service?
 - are there any special cases that the user needs to know about? do we expect each and every scenario that runs in a single cluster to work in the interdomain too?
 - how much are we bound to DNS? Can wee use other resolvers?
 - will IPv6 work?
 - is this verified in public clouds?
 - any expected performacne implications? for example do we have slower service resolution now, even in a single cluster?
 - future plans and next steps. what is still missing?